vm/src/obj/objdict.rs

```diff
@@ -650,36 +671,99 @@ macro_rules! dict_iterator {
                 vm.ctx.types.$iter_class.clone()
             }
         }
+
+        #[pyclass(module=false,name = $reverse_iter_class_name)]
+        #[derive(Debug)]
+        pub(crate) struct $reverse_iter_name {
+            pub dict: PyDictRef,
+            pub size: dictdatatype::DictSize,
+            pub position: AtomicCell<usize>,
+        }
+
+        #[pyimpl]
+        impl $reverse_iter_name {
+            fn new(dict: PyDictRef) -> Self {
+                $reverse_iter_name {
+                    position: AtomicCell::new(0),
+                    size: dict.size(),
+                    dict,
+                }
+            }
+
+            #[pymethod(name = "__next__")]
+            #[allow(clippy::redundant_closure_call)]
+            fn next(&self, vm: &VirtualMachine) -> PyResult {
+                if self.dict.entries.has_changed_size(&self.size) {
+                    return Err(
+                        vm.new_runtime_error("dictionary changed size during iteration".to_owned())
+                    );
+                }
+                let count = self.position.load() + 1;
+                match self.dict.len().checked_sub(count) {
+                    Some(mut pos) => {
+                        let (key, value) = self.dict.entries.next_entry(&mut pos).unwrap();
+                        self.position.store(count);
```

Comment:
@palaviv can you check the atomicity of this?

Reply:
I think @palaviv was busy for a while. Because I don't want to keep this PR open for too long time, here is a suggestion:

`count` looks like to be safely grow enough until overflow.
```rust
let count = self.position.fetch_add(1);
match ... {
    Some(..) => ... // we don't nees to store a new value here. it already was `fetch_add`ed. and this is the hot path
    None => { self.position.store(self.dict.len()+1); ... } // prevent overflow here
}
```

Reply:
I fixed this issue, but don't know it's what you expected.

Reply:
Yup